crypto/crypto_aes.c

```diff
@@ -29,9 +29,11 @@ aesnitest(uint8_t ptext[16], uint8_t * key, size_t keylen)
 	void * kexp_aesni;
 	uint8_t ctext_openssl[16];
 	uint8_t ctext_aesni[16];
+	int newkeylen;
 
 	/* Expand the key. */
-	AES_set_encrypt_key(key, keylen * 8, &kexp_openssl);
+	newkeylen = (int) (keylen * 8);
+	AES_set_encrypt_key(key, newkeylen, &kexp_openssl);
```

Comment:
This silences the warning, but doesn't really accomplish anything.  Also, `newkeylen` is unnecessary; you could just put the explicit cast into the function call.

I'd add an assertion (keylen < INT_MAX / 8, since that guarantees the conversion to int won't overflow) and then put the explicit cast into the function call.